draft-ietf-quic-transport.md

```diff
@@ -1084,10 +1084,10 @@ Sending a RETIRE_CONNECTION_ID frame indicates that the connection ID will not
 be used again and requests that the peer replace it with a new connection ID
 using a NEW_CONNECTION_ID frame.
 
-As discussed in {{migration-linkability}}, endpoints limit the use of a
-connection ID to a single network path where possible. Endpoints SHOULD retire
-connection IDs when no longer actively using the network path on which the
-connection ID was used.
+As discussed in {{migration-linkability}}, endpoints MUST limit the use of a
+connection ID to packets sent from a single local address.  Endpoints SHOULD
+retire connection IDs when they are no longer actively using the network path on
+which the connection ID was used.
```

Comment:
This text is now contradictory, because I can send from A -> B and A->C but the subsequent text now refers to "the network path"

Reply:
Perhaps the problem is the use of "network path". As I understand it, the requirement here is that from your perspective the 5-tuple be the same, right? The issue is that with NAT your perspective and others may different. So  perhaps "MUST limit the use of a connection ID to a single pair of remote/local address"

Reply:
> Perhaps the problem is the use of "network path".

I would agree, I think we've been trying to steer away from "network path" in favor of things from the perspective of each endpoint.

> The point about NAT rebinding/inadvertent migration here is not relevant and somewhat distracting.

The NAT rebinding is something that's been covered elsewhere already, I believe, but every review that we've had of this text has resulted in someone raising their hand to say "don't forget about NAT rebinding here", so have kept text in to settle that.

> The issue is that with NAT your perspective and others may different.

When we distilled this down, we went from the perspective of the requirements that are placed on each endpoint to help cope with this.

> This text is now contradictory, because I can send from A -> B and A->C

That shouldn't be the case: 
```
An endpoint MUST NOT reuse a connection ID when sending from more than one local
address, ...
```
```
Similarly, an endpoint MUST NOT reuse a connection ID when sending to more than
one destination address.
```

I think we can change the text above to not mention a "network path" and instead talk about the destination address, since that's what we really mean in this case.

Reply:
Updated the text in f7a9c3b to refer to being done with either your local address (migrated to a new address) or your destination address (only relevant for preferred_address and someday maybe server migration). Whenever you're done with an address on either end, you should probably retire the CID that you used with that address.